charts/tekton-pipelines/templates/webhook.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "tektonPipelines.fullname" . }}-webhook
+  labels:
+    app.kubernetes.io/component: webhook
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/name: webhook
+    app.kubernetes.io/part-of: tekton-pipelines
+    app.kubernetes.io/version: v0.36.0
+    pipeline.tekton.dev/release: v0.36.0
+    version: v0.36.0
+  {{- include "tektonPipelines.labels" . | nindent 4 }}
+spec:
+  maxReplicas: 5
+  metrics:
+  - resource:
+      name: cpu
+      targetAverageUtilization: 100
+    type: Resource
+  minReplicas: 1
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: tekton-pipelines-webhook
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "tektonPipelines.fullname" . }}-webhook
+  labels:
+    app: tekton-pipelines-webhook
+    app.kubernetes.io/component: webhook
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/name: webhook
+    app.kubernetes.io/part-of: tekton-pipelines
+    app.kubernetes.io/version: v0.36.0
+    pipeline.tekton.dev/release: v0.36.0
+    version: v0.36.0
+  {{- include "tektonPipelines.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.webhook.type }}
+  selector:
+    app.kubernetes.io/component: webhook
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/name: webhook
+    app.kubernetes.io/part-of: tekton-pipelines
+  {{- include "tektonPipelines.selectorLabels" . | nindent 4 }}
+  ports:
+	{{- .Values.webhook.ports | toYaml | nindent 2 -}} 
```

Comment:
not sure if this is correct

Reply:
I thought it looked weird too, but the webhook service rendered fine, see below. 

```apiVersion: v1
kind: Service
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"app":"tekton-pipelines-webhook","app.kubernetes.io/component":"webhook","app.kubernetes.io/instance":"tekton-pipelines","app.kubernetes.io/managed-by":"Helm","app.kubernetes.io/name":"tekton-pipelines","app.kubernetes.io/part-of":"tekton-pipelines","app.kubernetes.io/version":"v0.36.0","helm.sh/chart":"tekton-pipelines-0.2.4","pipeline.tekton.dev/release":"v0.36.0","version":"v0.36.0"},"name":"tekton-pipelines-webhook","namespace":"tekton-pipelines"},"spec":{"ports":[{"name":"http-metrics","port":9090,"targetPort":9090},{"name":"http-profiling","port":8008,"targetPort":8008},{"name":"https-webhook","port":443,"targetPort":8443},{"name":"probes","port":8080,"targetPort":0}],"selector":{"app.kubernetes.io/component":"webhook","app.kubernetes.io/instance":"tekton-pipelines","app.kubernetes.io/name":"tekton-pipelines","app.kubernetes.io/part-of":"tekton-pipelines"},"type":"ClusterIP"}}
  creationTimestamp: "2022-06-02T12:57:19Z"
  labels:
    app: tekton-pipelines-webhook
    app.kubernetes.io/component: webhook
    app.kubernetes.io/instance: tekton-pipelines
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: tekton-pipelines
    app.kubernetes.io/part-of: tekton-pipelines
    app.kubernetes.io/version: v0.36.0
    helm.sh/chart: tekton-pipelines-0.2.4
    pipeline.tekton.dev/release: v0.36.0
    version: v0.36.0
  name: tekton-pipelines-webhook
  namespace: tekton-pipelines
  resourceVersion: "17349"
  uid: 7890deaa-7126-49e7-8371-30616bdd5a8f
spec:
  clusterIP: 10.96.83.29
  clusterIPs:
  - 10.96.83.29
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - name: http-metrics
    port: 9090
    protocol: TCP
    targetPort: 9090
  - name: http-profiling
    port: 8008
    protocol: TCP
    targetPort: 8008
  - name: https-webhook
    port: 443
    protocol: TCP
    targetPort: 8443
  - name: probes
    port: 8080
    protocol: TCP
    targetPort: 8080
  selector:
    app.kubernetes.io/component: webhook
    app.kubernetes.io/instance: tekton-pipelines
    app.kubernetes.io/name: tekton-pipelines
    app.kubernetes.io/part-of: tekton-pipelines
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```